bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -104,7 +104,7 @@ def apply_prep(target_spec)
                 hook = inventory.plugins.get_hook(opts['plugin'], :puppet_library)
                 { 'target' => t,
                   'hook_proc' => hook.call(opts, t, self) }
-              rescue Bolt::Error => e
+              rescue Bolt::Error, Puppet::Error => e
```

Comment:
Specifically `Puppet::Module::Task::InvalidMetadata` was surfacing `JSON::ParserError` without making a `Bolt::Result` cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt apply -e "notice('hi')" -t custom_nodes
Starting: plan 
Starting: install puppet and gather facts on lgm13iwmk35hns2.delivery.puppetlabs.net
JSON::ParserError
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt apply -e "notice('hi')" -t custom_nodes
Starting: plan 
Starting: install puppet and gather facts on lgm13iwmk35hns2.delivery.puppetlabs.net
JSON::ParserError
```

Reply:
I think it should be `StandardError`

Reply:
Ok, for some reason I thought I remembered you and Lucy discussing how errors are handled, I figured that exclusively catching bolt errors was intentional here. Converting any `StandardError` to a bolt result makes sense. Updated.